pennylane/qchem/factorization.py

```diff
@@ -14,31 +14,107 @@
+    See theory section for more details.
+
+    For compressed double factorization (CDF), i.e., when ``compressed=True``, the above
+    decomposition is done by optimizing the following cost function :math:`\mathcal{L}`
+    in a greedy layered-wise manner:
+
+    .. math::
+
+       \mathcal{L}(U, Z) = \frac{1}{2} \bigg|V_{ijkl} - \sum_r^R \sum_{pq} U_{ip}^{(r)} U_{jp}^{(r)} Z_{pq}^{(r)} U_{kq}^{(r)} U_{lq}^{(r)}\bigg|_{\text{F}} + \rho \sum_r^R \sum_{pq} \bigg|Z_{pq}^{(r)}\bigg|^{\gamma},
+
+    where leaf tensors :math:`U` are defined by the antisymmetric orbital rotations :math:`X` such
+    that :math:`U^{(r)} = \exp{(X^{(r)})}`, :math:`|\cdot|_{\text{F}}` computes the Frobenius norm,
+    :math:`\rho` is a constant scaling factor, and :math:`\gamma` specifies the optional L1 and L2
+    regularization. See references `arXiv:2104.08957 <https://arxiv.org/abs/2104.08957>`__
+    and `arxiv:2212.07957 <https://arxiv.org/pdf/2212.07957>`__ for more details.
+
+    .. note::
+
+        Packages JAX and Optax are required when performing CDF with ``compressed=True``.
+        Install them using ``pip install jax optax``.
 
     Args:
         two_electron (array[array[float]]): two-electron integral tensor in the molecular orbital
-            basis arranged in chemist notation
-        tol_factor (float): threshold error value for discarding the negligible factors
-        tol_eigval (float): threshold error value for discarding the negligible factor eigenvalues
+            basis arranged in chemist notation.
+        tol_factor (float): threshold error value for discarding the negligible factors.
+            This will be used only when ``compressed=False``.
+        tol_eigval (float): threshold error value for discarding the negligible factor eigenvalues.
+            This will be used only when ``compressed=False``.
+        cholesky (bool): use Cholesky decomposition for obtaining the symmetric matrices
+            :math:`L^{(r)}` instead of eigendecomposition. Default is ``False``.
+        compressed (bool): use compressed double factorization to optimize the factors returned
+            in the decomposition. Look at the keyword arguments (``compression_kwargs``) for
+            the available options which must be provided only when ``compressed=True``.
+        regularization (string | None): type of regularization (``"L1"`` or ``"L2"``) to be
+            used for optimizing the factors. Default is to not include any regularization term.
+
+    Keyword Args:
+        num_factors (int): maximum number of factors that should be optimized for compressed
+            double factorization. Default is :math:`2\times N`, where `N` is the number of
+            dimensions of two-electron tensor.
+        num_steps (int): maximum number of epochs for optimizing each factor. Default is ``1000``.
+        optimizer (optax.optimizer): an optax optimizer instance. If not provided, `Adam
```

Comment:
```suggestion
        num_factors (int): Maximum number of factors that should be optimized for compressed
            double factorization. Default is :math:`2\times N`, where `N` is the number of
            dimensions of two-electron tensor.
        num_steps (int): Maximum number of epochs for optimizing each factor. Default is ``1000``.
        optimizer (optax.optimizer): An optax optimizer instance. If not provided, `Adam
```

Reply:
I am keeping all of them lowercase for uniformity. 👍

Reply:
Looks like we have some confusion and are going back and forth on this :sweat_smile: 
Let's go with the [PennyLane punctuation and spelling guide](https://docs.pennylane.ai/en/stable/development/guide/documentation.html#punctuation-and-spelling).

Reply:
The pennylane guide says to use proper punctuation for args which use more than a single line.